src/Store/StoreThread.re

```diff
@@ -187,7 +188,7 @@ let start =
     let workspaceUri =
       state.workspace
       |> Option.map((ws: Model.Workspace.workspace) => ws.workingDirectory)
-      |> Option.value(~default=Sys.getcwd())
+      |> OptionEx.value_lazy(~default=Sys.getcwd)
       |> Oni_Core.Uri.fromPath;
```

Comment:
Using a `switch` would make this simpler and clearer than adding a new utility function for lazy evaluation I think.

Reply:
Hmm, I tried this and have:
```
    let workspaceUri =
      state.workspace
      |> Option.map((ws: Model.Workspace.workspace) => ws.workingDirectory)
      |> (
        fun
        | None => Sys.getcwd()
        | Some(uri) => uri
      )
      |> Oni_Core.Uri.fromPath;
```

But I think this is a bit less readable - did you have something else in mind?

Otherwise, we do have other helper methods that are a similar level of abstraction to `OptionEx.value_lazy` - like [`or_lazy`](https://github.com/onivim/oni2/blob/master/src/Core/Utility/OptionEx.re#L36)

Reply:
I was thinking something more like this:

```suggestion
      switch (state.workspace) {
      | Some(workspace: Model.Workspace.workspace) => workspace.workingDirectory
      [ None => Sys.getcwd()
      }
      |> Oni_Core.Uri.fromPath;
```